packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -111,26 +109,27 @@ export function QuestionnaireForm(props: QuestionnaireFormProps): JSX.Element |
 
 interface QuestionnaireFormItemArrayProps {
   items: QuestionnaireItem[];
-  answers: Record<string, QuestionnaireResponseItemAnswer[]>;
+  responses: QuestionnaireResponseItem[];
   renderPages?: boolean;
   activePage?: number;
+  groupSequence?: number;
   onChange: (newResponseItems: QuestionnaireResponseItem[]) => void;
 }
 
 function QuestionnaireFormItemArray(props: QuestionnaireFormItemArrayProps): JSX.Element {
-  const [responseItems, setResponseItems] = useState<QuestionnaireResponseItem[]>(
+  const [currentResponseItems, setCurrentResponseItems] = useState<QuestionnaireResponseItem[]>(
     buildInitialResponseItems(props.items)
   );
 
   function setResponseItem(responseId: string, newResponseItem: QuestionnaireResponseItem): void {
-    const itemExists = responseItems.some((r) => r.id === responseId);
+    const itemExists = currentResponseItems.some((r) => r.id === responseId);
     let newResponseItems;
     if (itemExists) {
-      newResponseItems = responseItems.map((r) => (r.id === responseId ? newResponseItem : r));
+      newResponseItems = currentResponseItems.map((r) => (r.id === responseId ? newResponseItem : r));
```

Comment:
I will be pushing a follow-up PR for this specifically which is to merge the previous response with the newResponseItem. Since the responses are nested - we have to recursively look into the inputs and merge them with the new response. 

As of now, when we step back into a previous page, if we make changes to a group, and we press next, it will clear all other inputs in the group that wasn't changed at that moment.